### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by deancommasteven** [SoMuchForSubtlety/f1viewer#123]
Please check the [FAQ](https://github.com/SoMuchForSubtlety/f1viewer#Faq) and search [existing issues](https://github.com/SoMuchForSubtlety/f1viewer/issues?q=is%3Aissue) before you submit a new one!

**Describe the bug**
f1viewer 1.4 saved to and pulled credentials from the macOS keychain. 1.5 appears to be unable to see keychain and returns the message "ERROR: No keyring configured" when saving newly entered credentials. App functionality after logging in is unaffected.

**To Reproduce**
1. Open f1viewer 1.5 on macOS 
2. See that credentials are required even if previously saved in keychain
3. Enter valid credentials and select "save"
4. Successful log in with message in info pane: "ERROR: No keyring configured"
5. Subsequent app openings require manual credential entry

**Expected behaviour**
f1viewer should save to and read credentials from macOS keychain. Version 1.4 (redownloaded and tested) from github exhibits expected behavior.

**Screenshots**
<img width="1405" alt="Screen Shot 2021-03-24 at 9 08 42 PM" src="https://user-images.githubusercontent.com/25689810/112404087-0b9ae780-8ce6-11eb-98a1-3dc9ff0d04a3.png">

**F1TV account plan**
F1TV Pro, no VPN or proxy

**Desktop:**
 - macOS
 - Installed via brew (tested with direct package download from github, same result)
 - Version 1.5.0

**Logs**
f1viewer (via `brew upgrade`) was updated from 1.4 to 1.5 at 18:01:24. No other changes to app or system were made.
```
2021/03/24 17:44:04 [INFO] logged in!
2021/03/24 17:44:04 [INFO] checking for live session
2021/03/24 17:44:05 [INFO] New version found!
2021/03/24 17:44:05 [INFO] v1.5.0
2021/03/24 17:44:07 [INFO] no live session found
2021/03/24 17:46:39 [INFO] logged in!
2021/03/24 17:46:39 [INFO] checking for live session
2021/03/24 17:46:39 [INFO] New version found!
2021/03/24 17:46:39 [INFO] v1.5.0
2021/03/24 17:46:40 [INFO] no live session found
2021/03/24 17:47:40 [INFO] checking for live session
2021/03/24 17:47:41 [INFO] no live session found
2021/03/24 17:48:41 [INFO] checking for live session
2021/03/24 17:48:42 [INFO] no live session found
2021/03/24 17:49:42 [INFO] checking for live session
2021/03/24 17:49:45 [INFO] no live session found
2021/03/24 17:50:45 [INFO] checking for live session
2021/03/24 17:50:47 [INFO] no live session found
2021/03/24 17:51:47 [INFO] checking for live session
2021/03/24 17:51:50 [INFO] no live session found
2021/03/24 17:52:50 [INFO] checking for live session
2021/03/24 17:52:52 [INFO] no live session found
2021/03/24 17:53:52 [INFO] checking for live session
2021/03/24 17:53:55 [INFO] no live session found
2021/03/24 17:54:55 [INFO] checking for live session
2021/03/24 17:54:57 [INFO] no live session found
2021/03/24 17:55:57 [INFO] checking for live session
2021/03/24 17:56:00 [INFO] no live session found
2021/03/24 17:57:00 [INFO] checking for live session
2021/03/24 17:57:02 [INFO] no live session found
2021/03/24 17:58:02 [INFO] checking for live session
2021/03/24 17:58:05 [INFO] no live session found
2021/03/24 17:59:05 [INFO] checking for live session
2021/03/24 17:59:07 [INFO] no live session found
2021/03/24 18:00:07 [INFO] checking for live session
2021/03/24 18:00:10 [INFO] no live session found
2021/03/24 18:03:18 [ERROR] Could not access credential store: Specified keyring backend not available
2021/03/24 18:03:18 [ERROR] No keyring configured
2021/03/24 18:03:19 [INFO] checking for live session
2021/03/24 18:03:22 [INFO] no live session found
2021/03/24 18:03:45 [INFO] login successful!
2021/03/24 18:03:45 [ERROR] No keyring configured
2021/03/24 18:03:45 [INFO] detected active F1TV pro subscription
2021/03/24 18:04:22 [INFO] checking for live session
2021/03/24 18:04:24 [INFO] no live session found
2021/03/24 18:05:24 [INFO] checking for live session
2021/03/24 18:05:27 [INFO] no live session found
2021/03/24 18:06:27 [INFO] checking for live session
2021/03/24 18:06:30 [INFO] no live session found
```

**Comment by SoMuchForSubtlety** [comment]
Hmm, I can't reproduce the issue on a mac VM. Whats the output  if you run  `f1viewer -version`?

**Comment by deancommasteven** [comment]
Important detail I accidentally left out: I'm running the arm version for M1. Just checked the arm64 version and it accesses the keychain as expected on an M1 Mac. So it appears only the new arm version of 1.5 has this issue.

```
Version:     1.5.0
Git commit:  a7ac4e2a437f74b8fa02c9a0f318e81251f42834
Built:       2021-03-24T18:17:10Z
```

**Comment by SoMuchForSubtlety** [comment]
There is probably an issue with cross compiling CGO for a different CPU architecture.

I don't have access to an m1 mac to test on myself. Can you install the go compiler https://golang.org/doc/install and then run 
```go install github.com/SoMuchForSubtlety/f1viewer@latest```, that will compile f1viewer from source. I expect that the resulting binary will be able to access the keychain.

**Comment by deancommasteven** [comment]
Confirmed that compiling from source on my M1 Mac exhibits expected behavior. v1.5.0 is able to access the keychain.

**Comment by deancommasteven** [comment]
Also confirmed that compiling from source with the provided command compiles the amd64 version of the app, which was confirmed working previously. How do I specify to compile the arm64 version locally?

**Comment by deancommasteven** [comment]
More info:

I was unable to get the compile mentioned above to work with the supplied command, `go install github.com/SoMuchForSubtlety/f1viewer@latest` which returned the error: `can't load package: package github.com/SoMuchForSubtlety/f1viewer@v1.5.0: can only use path@version syntax with 'go get'`

I also had to enable modules with `export GO111MODULE=on`, before running the `go get` command to compile. 

This is what resulted in the Intel binary being compiled.

**Comment by SoMuchForSubtlety** [comment]
What go version are you using? go 1.16 added support for m1 and  `go install package@version`

**Comment by deancommasteven** [comment]
Doh. I was on an old version of go. Updated to `go1.16.2 darwin/arm64` and ran the `go install` command and received the following:

```
go install github.com/SoMuchForSubtlety/f1viewer@latest: github.com/SoMuchForSubtlety/f1viewer@v1.5.0
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
```

Is there something I should do to allow the install to proceed? I was unable to find a solution that could be completed on my end for this particular issue.

**Comment by SoMuchForSubtlety** [comment]
Oh, looks like `go install` does not support replace directives. `go get github.com/SoMuchForSubtlety/f1viewer` should work though.

**Comment by deancommasteven** [comment]
Ok, got the arm64 v1.5.0 compiled locally. It can access the keychain as you expected.

**Comment by SoMuchForSubtlety** [comment]
Thanks for testing that.

I will try to get CGO cross compilations working, but I don't have high hopes.

**Comment by deancommasteven** [comment]
Fair enough, thank you for looking into it. For now I'll just compile the arm version from source for future updates. I'd be happy to test builds on my Mac. 

Great app, by the way, big fan.
